service/src/test/java/uk/gov/hmcts/reform/fpl/utils/CaseDetailsHelperTest.java

```diff
@@ -136,6 +141,100 @@ void shouldRemoveTempFieldsBasedOfGivenClass2() {
         }
     }
 
+    @Nested
+    class AddFields {
+
+        private Map<String, Object> data = new HashMap<>();
+
+        private CaseDetailsMap caseDetails;
+
+        @BeforeEach
+        void populateMap() {
+            data.put("key1", "some value 1");
+            data.put("key2", "some value 2");
+
+            caseDetails = caseDetailsMap(CaseDetails.builder().data(data).build());
+        }
+
+        @Test
+        void shouldNotUpdateAnyFieldsWhenFieldsGroupIsNotSpecified() {
+
+            final TestClass target = new TestClass("one", "two", "three", "four");
+
+            final CaseDetailsMap actualCaseDetails = addFields(caseDetails, target);
+
+            assertThat(actualCaseDetails).containsExactly(
+                entry("key1", "some value 1"),
+                entry("key2", "some value 2"));
+        }
+
+        @Test
+        void shouldNotUpdateAnyFieldsWhenFieldsGroupDoesNotExistsInTargetObject() {
+
+            final TestClass target = new TestClass("one", "two", "three", "four");
+
+            final CaseDetailsMap actualCaseDetails = addFields(caseDetails, target, "Non existing");
+
+            assertThat(actualCaseDetails).containsExactly(
+                entry("key1", "some value 1"),
+                entry("key2", "some value 2"));
+        }
+
+        @Test
+        void shouldAddFieldsFromSingleGroupInTargetObject() {
+
+            final TestClass target = new TestClass("one", "two", "three", "four");
+
+            final CaseDetailsMap actualCaseDetails = addFields(caseDetails, target, "Group1");
+
+            assertThat(actualCaseDetails).containsExactly(
+                entry("key1", "one"),
+                entry("key2", "some value 2"),
+                entry("key3", "three")
+            );
+        }
+
+        @Test
+        void shouldAddFieldsFromMultipleGroupsInTargetObject() {
+
+            final TestClass target = new TestClass("one", "two", "three", "four");
+
+            final CaseDetailsMap actualCaseDetails = addFields(caseDetails, target, "Group1", "Group2");
+
+            assertThat(actualCaseDetails).containsExactly(
+                entry("key1", "one"),
+                entry("key2", "two"),
+                entry("key3", "three"));
+        }
+
+        @Test
+        void shouldRemoveFieldsFromSingleGroupWhenNullInTargetObjects() {
+
+            final TestClass target = new TestClass(null, null, null, null);
+
+            final CaseDetailsMap caseDetailsMap = addFields(caseDetails, target, "Group2");
+
+            assertThat(caseDetailsMap).containsExactly(entry("key1", "some value 1"));
+        }
+
+        @Test
+        void shouldRemoveFieldsFromMultipleGroupsWhenNullInTargetObjects() {
+
+            final TestClass target = new TestClass(null, null, null, null);
+
+            final CaseDetailsMap actualCaseDetails = addFields(caseDetails, target, "Group1", "Group2");
```

Comment:
I wonder if "putFields" would be a better name for this method (since it replaces existing fields, like the "put" method in a map).

Reply:
Or even "copyFieldsMarkedWithGivenGroups".